cmd/juju/application/updateseries.go

```diff
@@ -108,33 +117,56 @@ func (c *updateSeriesCommand) Init(args []string) error {
 
 // Run implements cmd.Run.
 func (c *updateSeriesCommand) Run(ctx *cmd.Context) error {
-
-	if c.updateSeriesClient == nil {
-		apiRoot, err := c.NewAPIRoot()
+	var apiRoot api.Connection
+	if c.updateMachineSeriesClient == nil && c.updateApplicationSeriesClient == nil {
+		var err error
+		apiRoot, err = c.NewAPIRoot()
 		if err != nil {
 			return errors.Trace(err)
 		}
 		defer apiRoot.Close()
-		c.updateSeriesClient = application.NewClient(apiRoot)
-		defer c.updateSeriesClient.Close()
 	}
 
-	if c.updateSeriesClient.BestAPIVersion() < 5 {
-		return errors.New("updating the application series is not supported by this API server")
+	if c.machineNumber != "" {
+		if c.updateMachineSeriesClient == nil {
+			c.updateMachineSeriesClient = machinemanager.NewClient(apiRoot)
+			defer c.updateMachineSeriesClient.Close()
+		}
+		if c.updateMachineSeriesClient.BestAPIVersion() < 4 {
+			return errors.New("updating the machine series is not supported by this API server")
+		}
+		return c.updateMachineSeries()
 	}
 
-	if c.machineNumber != "" {
-		// TODO(hml) 2017/08/14
-		// implment updateMachineSeries
-		return errors.NotSupportedf("update-series by machine name is")
+	if c.applicationName != "" {
+		if c.updateApplicationSeriesClient == nil {
+			c.updateApplicationSeriesClient = application.NewClient(apiRoot)
+			defer c.updateApplicationSeriesClient.Close()
+		}
+		if c.updateApplicationSeriesClient.BestAPIVersion() < 5 {
+			return errors.New("updating the application series is not supported by this API server")
+		}
+		return c.updateApplicationSeries()
 	}
 
-	return c.updateApplicationSeries(ctx)
+	// This should never happen...
+	return errors.New("programmer error, please file a bug")
```

Comment:
Let's make this something like "no machine or application specified"

Reply:
changed.